src/core/src/parallel/shortcuts.rs

```diff
@@ -15,13 +19,34 @@ use rayon::iter::{Map, MapFn};
 /// let s = (0..100_i32).par_map(|i| -i).sum();
 /// assert_eq!(-4950, s);
 /// ```
-pub trait ParallelShortcuts : IntoParallelIterator + Sized {
+pub trait ParallelShortcuts: Sized {
+    /// The iterator type
+    type Iter: ParallelIterator;
+
+    /// Dummy function to make it work
+    fn _into_par_iter(self) -> Self::Iter;
```

Comment:
We don't know if we will use it later, that's what YAGNI is about ;-). And it feels wrong to have a "dummy" function in a trait, just for the sake of it too =)